espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/Screenshot.java

```diff
@@ -0,0 +1,88 @@
+package io.appium.espressoserver.lib.handlers;
+
+import android.graphics.Bitmap;
+import android.os.Environment;
+import android.util.Base64;
+import android.view.View;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.model.AppiumParams;
+import io.appium.espressoserver.lib.viewaction.RootViewFinder;
+
+public class Screenshot implements RequestHandler<AppiumParams, String> {
+
+    @Override
+    public String handle(AppiumParams params) throws AppiumException {
+        return takeScreenshot();
+    }
+
+
+    private String takeScreenshot() throws AppiumException {
```

Comment:
I feel this method is doing some extra work. Cannot we just save the screenshot to a string/memory stream and do all the transformations in memory instead of creating a file and asking android.permission.WRITE_EXTERNAL_STORAGE permissions?